src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -375,41 +375,58 @@ public bool TrySkip()
                 return true;
             }
 
-            return TrySkipHelper();
+            Utf8JsonReader restore = this;
+            bool success = TrySkipPartial(targetDepth: CurrentDepth);
+            if (!success)
+            {
+                // Roll back the reader if it contains partial data.
+                this = restore;
+            }
+
+            return success;
         }
 
-        private bool TrySkipHelper()
+        /// <summary>
+        /// Tries to skip the children of the current JSON token, advancing the reader even if there is not enough data.
+        /// The skip operation can be resumed later, provided that the same <paramref name="targetDepth" /> is passed.
+        /// </summary>
+        /// <param name="targetDepth">The target depth we want to eventually skip to.</param>
+        /// <returns>True if the entire JSON value has been skipped.</returns>
+        internal bool TrySkipPartial(int targetDepth)
```

Comment:
The existing `TrySkipHelper` has been retrofitted to the resumable `TrySkipPartial` method, minus the checkpointing semantics.